src/Workspaces/Core/Portable/Recommendations/AbstractRecommendationService.cs

```diff
@@ -29,6 +29,26 @@ internal abstract class AbstractRecommendationService : IRecommendationService
             return symbols;
         }
 
+        protected static IEnumerable<ISymbol> GetRecommendedNamespaceNameSymbols(
+            SemanticModel semanticModel, SyntaxNode declarationSyntax, CancellationToken cancellationToken)
+        {
+            var containingNamespaceSymbol = semanticModel.Compilation.GetCompilationNamespace(
+                semanticModel.GetEnclosingNamespace(declarationSyntax.SpanStart, cancellationToken));
+
+            var symbols = semanticModel.LookupNamespacesAndTypes(declarationSyntax.SpanStart, containingNamespaceSymbol)
+                                       .Where(recommendationSymbol => IsNonIntersectingNamespace(recommendationSymbol, declarationSyntax));
+
+            return symbols;
+        }
+
+        protected static bool IsNonIntersectingNamespace(ISymbol recommendationSymbol, SyntaxNode declarationSyntax)
+        {
+            return recommendationSymbol.IsNamespace() &&
```

Comment:
```
Namespace Foo$$
End Namespace
```
In the above `Foo` shouldn't be considered for the completion list unless it's also declared somewhere else. This applies to VB and C#.

```
Namespace Foo.$$
    Namespace Bar
    End Namespace
End Namespace
```
Here, `Bar` shouldn't be included the completion, because it will always be a child of whatever the user is about to type (again, unless it's also declared somewhere else). For whatever reason this only applies to how VB presents the symbols. Regardless, I added tests for both languages.